redbeat/schedulers.py

```diff
@@ -174,3 +188,36 @@ def schedule(self):
             d[entry.name] = entry
 
         return d
+
+    def tick(self, **kwargs):
+        if self.lock:
+            logger.debug('beat: Extending lock...')
+            rdb.pexpire(self.lock_key, int(self.lock_timeout * 1000))
+        return super(RedBeatScheduler, self).tick(**kwargs)
+
+    def close(self):
+        if self.lock:
```

Comment:
This only ever gets called from within beat? What happens on a dyno recycle will beat have a graceful shutdown and this get called?